synapseutils/migrate_functions.py

```diff
@@ -290,79 +399,183 @@ def _wait_futures(conn, cursor, futures, return_when, continue_on_error):
 
-    executor, max_concurrent_file_copies = _get_executor()
+    _verify_storage_location_ownership(syn, storage_location_id)
 
     test_import_sqlite3()
     import sqlite3
     with sqlite3.connect(db_path) as conn:
 
         cursor = conn.cursor()
         _ensure_schema(cursor)
+
+        _verify_index_settings(
+            cursor,
+            db_path,
+            root_id,
+            storage_location_id,
+            file_version_strategy,
+            include_table_files
+        )
         conn.commit()
 
-        entity = syn.get(entity, downloadFile=False)
-        if not _check_indexed(cursor, entity):
+        entity = syn.get(root_id, downloadFile=False)
+        try:
             _index_entity(
                 conn,
                 cursor,
                 syn,
                 entity,
                 None,
+                storage_location_id,
                 file_version_strategy,
-                create_table_snapshot,
+                include_table_files,
                 continue_on_error,
             )
 
+        except _IndexingError as indexing_ex:
+            logger.exception(
+                "Aborted due to failure to index entity %s of type %s. Use the continue_on_error option to skip "
+                "over entities due to individual failures.",
+                indexing_ex.entity_id,
+                indexing_ex.concrete_type,
+            )
+
+            raise indexing_ex.__cause__
+
+    return MigrationResult(syn, db_path)
+
+
+def _confirm_migration(cursor, force, storage_location_id):
+    # we proceed with migration if either using the force option or if
+    # we can prompt the user with the count if items that are going to
+    # be migrated and receive their confirmation from shell input
+    confirmed = force
+    if not force:
+        count = cursor.execute(
+            "select count(*) from migrations where status = ?",
+            (_MigrationStatus.INDEXED.value,)
+        ).fetchone()[0]
+
+        if count == 0:
+            logger.info("No items for migration.")
+
+        elif sys.stdout.isatty():
+            uinput = input("{} items for migration to {}. Proceed? (y/n)? ".format(
+                count,
+                storage_location_id
+            ))
+            confirmed = uinput.strip().lower() == 'y'
+
+        else:
+            logger.info(
+                "%s items for migration. "
+                "force option not used, and console input not available to confirm migration, aborting. "
+                "Use the force option or run from an interactive shell to proceed with migration.",
+                count
+            )
+
+    return confirmed
+
+
+def migrate_indexed_files(
+    syn: synapseclient.Synapse,
+    db_path: str,
+    create_table_snapshots=True,
+    continue_on_error=False,
+    force=False
+):
+    """
+    Migrate files previously indexed in a sqlite database at the given db_path using the separate
```

Comment:
Add to doc warning that if the entity changes between the time of the index and the time of migration then changes may not be reflected in the migration.